src/shared_state/conflicts.md

```diff
@@ -0,0 +1,52 @@
+# Detecting conflicts
+
+Modifications to shared state can be considered conflicting when two conditions are met:
+
+1. They affect an overlapping section of a data structure, such as if two users both try to set the title of the same todo-list item
+
+2. They are [concurrent](https://en.wikipedia.org/wiki/Concurrency_(computer_science)) - meaning that the participants in the system cannot conclusively determine the order in which they occurred
+
+_Concurrent updates which affect unrelated data_ can be resolved without issue, as can _sequential updates to related data_.
+
+Concurrency can effectively be ignored from a client's perspective in schemes that use a central server as a single source of truth.
+In practice one message will always arrive slightly before another, and even if they were truly simultaneous the server could arrange them into an arbitrary order.
+Because this can be done for all messages, all updates can be said to have a [total order](https://en.wikipedia.org/wiki/Total_order), which makes it possible for all clients to compute the same results when they are applied sequentially.
+
+## Conflict resolution in webxdc applications
+
+Many webxdc applications are incapable of producing conflicts, either because they do not use the network communication APIs, or because any given data structure can only have one writer.
```

Comment:
```suggestion
Many webxdc applications are incapable of producing conflicts, either because they do not use the network communication APIs, or because any given data structure can only have one writer.  For example, many of the available [webxdc games](https://webxdc.org/apps/) offer a high-score table and how users scored will typically be reported back to the chat. Such a gaming app will simply post their user's highscore and everybody will collect all arriving highscore data and show it to their users in a sorted list. There is hardly any problem if a message is lost or messages get re-ordered.  Eventually everybody will see roughly the same highscore list. 
```